.github/workflows/facebook.yml

```diff
@@ -3,31 +3,52 @@ name: Build Facebook plug
 on:
   push:
     tags:
-      - 'facebook-v*.*.**'
+      - 'facebook-v**'
+    branches:
+      - '**'
+    paths:
+      - 'build.sbt'
+      - 'dataplug-facebook/**'
+      - '.github/workflows/facebook.yml'
+
+env:
+  DATAPLUG: dataplug-facebook
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
-    env:
-      DATAPLUG: dataplug-facebook
-      REPOSITORY: dataplug-facebook
     steps:
       - uses: actions/checkout@v2
+
       - name: Setup Java environment
         uses: actions/setup-java@v1
         with:
           java-version: 11
           java-package: jdk
           architecture: x64
-      - name: Setup NodeJs environment
-        uses: actions/setup-node@v1
-        with:
-          node-version: '10.x'
-      - name: Docker stage
+
+      - name: Container Registry Login
+        run: docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
+
+      - name: Container Image Prepare
         run: sbt "project $DATAPLUG" docker:stage
-        if: github.event_name == 'push'
-      - name: Build, tag, and push image to docker hub
+
+      - name: Container Image Build
+        run: docker build -t hubofallthings/$DATAPLUG:$GITHUB_SHA $DATAPLUG/target/docker/stage/
+
+      - name: Container Image Push
+        run: docker push hubofallthings/$DATAPLUG:$GITHUB_SHA
+
+      - name: Container Image Push - Master
+        if: startsWith(github.ref, 'refs/heads/master')
+        run: |
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:master
+          docker push hubofallthings/$DATAPLUG:master
+
+      - name: Container Image Push - Tag
+        if: startsWith(github.ref, 'refs/tags')
         run: |
-          docker build -t hubofallthings/$REPOSITORY $DATAPLUG/target/docker/stage/
-          docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
-          docker push hubofallthings/$REPOSITORY
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker push hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:latest
+          docker push hubofallthings/$DATAPLUG:latest
```

Comment:
I thought `latest` was to be removed?

Reply:
I would like to keep `latest` tag as an additional tag and use it only as a cache for future builds.